pkg/fsm/fsm_active_file_io.go

```diff
@@ -95,7 +95,7 @@ func (f *FsMachine) readFile(file *types.OutputFile) StateFn {
 	fi, err := os.Stat(sourcePath)
 	if err != nil {
 		file.Response <- &types.Event{
-			Name: types.EventNameReadFailed,
+			Name: types.EventNameFileNotFound,
 			Args: &types.EventArgs{"err": fmt.Errorf("failed to stat %s, error: %s", file.Filename, err)},
 		}
 		return backoffState
```

Comment:
Not a big issue,  but you could do something more specific on the err returned to make sure it's a 'file not found' error.
```
if _, err := os.Stat("/path/to/whatever"); os.IsNotExist(err) {
	// path/to/whatever does not exist
}
```
As the err returned by os.Stat is of a generic type, I think it's a *PathError

Reply:
Good point!